### PR TITLE
Update UI for group chat creation and profile

### PR DESCRIPTION
**Pull request by Ferossgp**
https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=8136%3A270

### CODE REVIEW

**Comment by errorists** [comment]
some mistakes

IMG_1346

• same as with other chats it should use modal presentation, 'X' to close
• bottom toolbar, the thin separator border should use the `Light Grey` colour
• Search is missing the ['Cancel' button to dismiss ](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=8471%3A38086)

IMG_1347

• the button to confirm chat creation should be without the chevron and use the '500' font weight

**Comment by errorists** [comment]
Profiles need some additional work to be complete,

• as an admin, [no way to edit the chat name](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=6130%3A0)
• no differentiation between deleting the chat as an admin and leaving it. Deleting, removes the chat for all members, leaving removes the chat only for you.
• Delete chat confirmation alert has wrong text and button styling [Figma](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=480%3A103)
• There should be a 16px margin between the ••• icon / `Admin` text label and the text string that's next to it
• top bar is missing the bottom divider and scrolling behaviour, the ui should mirror the one we have in the profile tab
• Add members view should be presented modally, is missing its title, and the bottom toolbar has [incorrect styling](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=259%3A6939)
• Add members, when the list is empty it's showing that I don't have any contacts yet which is not true. if the list is empty, instead it should rather say that `All your contacts are already in the group` [Figma](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=8738%3A1365)

IMG_1348

**Comment by errorists** [comment]
the first screen of creating a group chat where you select members: 

- selecting search, animates up the toolbar but not enough to make it appear above the keyboard
- search is missing the same 'cancel' to dismiss button we have in chat tab 

screen where you name the group
- no way to dismiss the keyboard
- keyboard covers the toolbar, so i was unable to proceed forward without tapping return to dismiss

I thought we killed the 'Delete chat' buttons? what does it do and why it's available for both admins and members alike?

Edit chats screen
- done button should use the blue colour as long as requirements are met (>=1 character)

after bringing the app from background, I'm stuck on a spinner centred in the chats screen, all other views work ok.

**Comment by cammellos** [comment]
> I thought we killed the 'Delete chat' buttons? what does it do and why it's available for both admins and members alike?

Delete chat makes the member (or admin) leave the chat & delete locally
While Leave chat leaves the chat but it's still open (so user can read old messages)

**Comment by errorists** [comment]
Yea this doesn't make sense to me xD I'd only leave 'Leave chat' with the behaviour of deleting so `Leave chat would delete the chat and remove it from list locally`

**Comment by errorists** [comment]
IMG_1410
can't seem to get rid of that, no idea if it's caused by something in this PR or not. happened when I opened the app from background

edit: managed to reproduce it after a clean re-install

**Review by errorists — APPROVED** [review]

File: src/status_im/group_chats/core.cljs [hunk 1]
```diff
@@ -98,33 +98,26 @@
...
-  [{:keys [db] :as cofx}]
-  (let [new-name (get-in cofx [:db :group-chat-profile/profile :name])
-        current-chat-id (:current-chat-id db)]
-    (when (valid-name? new-name)
-      {::json-rpc/call [{:method (json-rpc/call-ext-method (waku/enabled? cofx) "changeGroupChatName")
-                         :params [nil current-chat-id new-name]
-                         :on-success #(re-frame/dispatch [::chat-updated %])}]})))
+  {:events [:group-chats.ui/name-changed]}
+  [{:keys [db] :as cofx} chat-id new-name]
+  (println new-name chat-id)
```
Comment (flexsurfer) [inline src/status_im/group_chats/core.cljs]: leftover

File: src/status_im/ui/screens/group/subs.cljs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+(ns status-im.ui.screens.group.subs
```
Comment (flexsurfer) [inline src/status_im/ui/screens/group/subs.cljs]: all subs should be in one ns, root subs ns

File: src/status_im/ui/screens/group/views.cljs [hunk 3]
```diff
@@ -65,127 +54,221 @@
...
                           :key-fn                       :address
                           :render-fn                    render-contact
                           :bounces                      false
                           :keyboard-should-persist-taps :always
                           :enable-empty-sections        true}]]]
-       [bottom-container {:on-press #(re-frame/dispatch [:group-chats.ui/create-pressed group-name])
-                          :disabled (string/blank? group-name)
-                          :label (i18n/label :t/create-group-chat)
-                          :accessibility-label :create-group-chat-button}]])))
+       [react/view {:style (styles/bottom-container)}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/group/views.cljs]: we have `status-im.ui.components.toolbar` component

File: src/status_im/ui/screens/profile/user/sheet/views.cljs [hunk 4]
```diff
@@ -42,3 +42,7 @@
          :accessibility-label :group-info-button
          :icon                :main-icons/delete
          :on-press            #(hide-sheet-and-dispatch [:my-profile/remove-current-photo])}])]))
+
+(defn edit-group-chat-name []
```
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/user/sheet/views.cljs]: ?

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
@Ferossgp 
## ISSUE 1: Tap back from "Invite" screen or creating group chat navigates to white screen (Android Only)
photo_2020-04-06 16 03 10
### ISSUE 2: "Decline invitation"  is not tappable
### ISSUE 3:  If tap quickly 2 times on "Create group chat" button - two chats are created
### ISSUE 4: Not obvious from group info screen how many users are admins.
I mean there is no mark / distinction for the user who has been made admin.
Monosnap 2020-04-06 16-19-52
### ISSUE 5: "Unknown"  invited you to group chat message when creating group chat
From receiver side until join:
Monosnap 2020-04-06 16-30-31


@Ferossgp let me know if smth should be fixed separately. 
Thanks!

**Comment by Ferossgp** [comment]
@cammellos can you please have a look at Issue 5?

**Comment by cammellos** [comment]
@churik I have fixed point 5, it should be ready to be tested again as @Ferossgp addressed the other points. Thanks!

**Comment by churik** [comment]
All issues from https://github.com/status-im/status-react/pull/10099#issuecomment-609834193 are fixed, I'm starting refactoring e2e tests for group chats.
@errorists can you please take a look, is it good enough from your POV?

**Comment by churik** [comment]
@Ferossgp 
one more small issue (overlapping on New group chat screen):
Monosnap 2020-04-08 11-17-58

Also I can't scroll to the end of the list with opened keyboard if I have 10 members - last 4 are overlapped with keyboard.

**Comment by Ferossgp** [comment]
@churik fixed the list

**Comment by churik** [comment]
tests results are here: https://github.com/status-im/status-react/pull/10315

**Pull request merged**